column-profiling/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -0,0 +1,126 @@
+import com.r3.conclave.common.EnclaveInstanceInfo;
+import com.r3.conclave.mail.Curve25519PrivateKey;
+import com.r3.conclave.mail.EnclaveMail;
+import com.r3.conclave.mail.PostOffice;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.security.PrivateKey;
+import java.util.ArrayList;
+import java.io.*;
+import java.util.UUID;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Output;
+import com.r3.conclave.sample.dataanalysis.common.DataSetSerializer;
+import com.r3.conclave.shaded.kotlin.Pair;
+import com.r3.conclave.sample.dataanalysis.common.UserProfile;
+import com.r3.conclave.sample.dataanalysis.common.DataSet;
+
+
+
+public class Client {
+        private static Kryo kryo;
+
+        public static void main(String[] args) throws Exception {
+                // This is the client that will upload secrets to the enclave for processing.
+
+                kryo = new Kryo();
+                if (args.length == 0) {
+                        System.err.println("Please pass the user data for processing as argument to the command line using --args=\"data fields with space separation \"");
+                        return;
+                }
+                ArrayList<UserProfile> l = new ArrayList<UserProfile>(){};
+                String constraint = args[0];
+                for (int i =1;i< args.length; i = i+4){
+                       UserProfile u = new UserProfile(args[i],
+                               Integer.parseInt(args[i+1]),
+                               args[i+2],
+                               args[i+3]);
+                       l.add(u);
+
+                }
+
+                // Connect to the host, it will send us a remote attestation (EnclaveInstanceInfo).
+                // Establish a TCP connection with the host
+                Pair<DataInputStream, DataOutputStream> streams = establishConnection();
+                DataInputStream fromHost = streams.getFirst();
+                DataOutputStream toHost = streams.getSecond();
+
+                //Verify attestation here
+                EnclaveInstanceInfo attestation = verifyAttestation(fromHost, constraint);
+
+                // Check it's the enclave we expect. This will throw InvalidEnclaveException if not valid.
+                System.out.println("Connected to " + attestation);
+                // Two distinct signing key hashes can be accepted.
+                // Release mode: 360585776942A4E8A6BD70743E7C114A81F9E901BF90371D27D55A241C738AD9
+                // Debug mode:   4924CA3A9C8241A3C0AA1A24A407A999A86401D2B79FA9FF84932DA798A942166D4
+                EnclaveConstraint.parse("S:360585776942A4E8A6BD70743E7C114A81F9E901BF90371D27D55A241C738AD9 "
+                        + "S:4924CA3A9C8241A3C0AA1A24A407AA86401D2B79FA9FF84932DA798A942166D4 PROD:1 SEC:INSECURE").check(attestation);
```

Comment:
You're parsing the constraint string but then not using the constraint object that's produced.

Reply:
done